jsapp/xlform/src/model.inputParser.coffee

```diff
@@ -94,25 +94,52 @@ module.exports = do ->
     _curGrp().export().__rows
 
   inputParser.parseArr = parseArr
-  inputParser.parse = (o)->
-    translations = o.translations
-    if o['#active_translation_name']
-      _existing_active_translation_name = o['#active_translation_name']
-      delete o['#active_translation_name']
 
+  # pass baseSurvey whenever you import other asset into existing form
+  inputParser.parse = (o, baseSurvey)->
+    translations = o.translations
     if translations
-      if translations.indexOf(null) is -1 # there is no unnamed translation
-        if _existing_active_translation_name
-          throw new Error('active translation set, but cannot be found')
-        o._active_translation_name = translations[0]
-        translations[0] = null
-      else if translations.indexOf(null) > 0
-        throw new Error("""
-                        There is an unnamed translation in your form definition.
-                        Please give a name to all translations in your form.
-                        """)
-      else if _existing_active_translation_name # there is already an active null translation
-        o._active_translation_name = _existing_active_translation_name
+      if translations.length > 1 and translations.indexOf(null) isnt -1
+        throw new Error("There is an unnamed translation in your form definition. Please give a name to all translations in your form.")
+
+      ###
+      TRANSLATIONS HACK (Part 1/2):
+      all the coffee code assumes first language to be null, and we don't want
+      to introduce potential code-breaking refactor in old code, so we store
+      first language, then replace with null and reverse this just before saving
+      NOTE: when importing assets from Library into form, we need to make sure
+      the default language is the same (or force baseSurvey default language)
+      ###
+      if baseSurvey
+        formDefaultLang = baseSurvey._initialParams.translations_0 or null
+        translatedProp = o.translated[0]
```

Comment:
Why just `o.translated[0]`? If I have the following library question:
```json
{
    "settings": {},
    "translations": [
        "a (a)",
        "b"
    ],
    "choices": [
        {
            "$kuid": "EMWg8TEsX",
            "list_name": "thingy",
            "$autovalue": "choice_name",
            "name": "choice_name",
            "label": [
                "a lang c label",
                "b lang c label"
            ]
        }
    ],
    "survey": [
        {
            "select_from_list_name": "thingy",
            "name": "q_name",
            "constraint_message": [
                "a lang constr msg",
                "b lang constr msg"
            ],
            "hint": [
                "a lang hint",
                "b lang hint"
            ],
            "$autoname": "q_name",
            "required": false,
            "label": [
                "a lang q label",
                "b lang q label"
            ],
            "required_message": [
                "a lang req",
                "b lang req"
            ],
            "$kuid": "went0vIoQ",
            "type": "select_one"
        }
    ],
    "translated": [
        "constraint_message",
        "hint",
        "label",
        "required_message"
    ],
    "schema": "1"
}
```
...and I add it to a survey whose default language is `b`, then I end up with my question label being `a lang q label`, **but** the choice label shown is `b lang c label`:
![image](https://user-images.githubusercontent.com/2085013/45274737-ca73bc80-b486-11e8-928e-35725ca305d2.png)

Debugging shows that `o.translated` is `["constraint_message", "hint", "label", "required_message"]`.